backend/de.metas.acct.base/src/main/java-legacy/org/compiere/acct/Doc_MatchPO.java

```diff
@@ -192,7 +193,9 @@ private void updateFromPurchaseOrderLine(final FactLine factLine)
 		}
 
 		final I_C_OrderLine orderLine = docLine.getOrderLine();
-		factLine.setC_BPartner_ID(orderLine.getC_BPartner_ID());
+
+		final BPartnerLocationId bPartnerLocationId = BPartnerLocationId.ofRepoId(orderLine.getC_BPartner_ID(), orderLine.getC_BPartner_Location_ID());
+		factLine.setC_BPartner_Location_ID(bPartnerLocationId);
```

Comment:
hmm, not sure but i think that setFromDimension from below would clear ur C_BPartner_Location_ID....

wondering why not introducing BPartnerLocationid too in Dimension? won't be much more solid?

Reply:
> hmm, not sure but i think that setFromDimension from below would clear ur C_BPartner_Location_ID

it won't, the location is cleared only if the bpartner has really changed see: `org.adempiere.acct.api.impl.FactAcctBL#setC_BPartner_ID`

```JAVA

	public void setC_BPartner_ID(@NonNull final I_Fact_Acct factAcct, @Nullable final BPartnerId bpartnerId)
	{
		final BPartnerId currentBPartnerId = BPartnerId.ofRepoIdOrNull(factAcct.getC_BPartner_ID());

		if (BPartnerId.equals(currentBPartnerId, bpartnerId))
		{
			return;
		}

		factAcct.setC_BPartner_ID(BPartnerId.toRepoId(bpartnerId));
		factAcct.setC_BPartner_Location_ID(-1);
	}

```